src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js

```diff
@@ -0,0 +1,533 @@
+import React, { Fragment } from 'react';
+import DownCaretIcon from 'react-icons/lib/fa/caret-down';
+
+import {
+  find,
+  isEmpty,
+  maxBy,
+  reject,
+  sortBy,
+} from 'lodash';
+import { Row, Column } from '@ncigdc/uikit/Flex';
+import Button from '@ncigdc/uikit/Button';
+import { Tooltip } from '@ncigdc/uikit/Tooltip';
+import { visualizingButton, zDepth1 } from '@ncigdc/theme/mixins';
+import EntityPageHorizontalTable from '@ncigdc/components/EntityPageHorizontalTable';
+import Dropdown from '@ncigdc/uikit/Dropdown';
+import DropdownItem from '@ncigdc/uikit/DropdownItem';
+import Hidden from '@ncigdc/components/Hidden';
+import DownloadVisualizationButton from '@ncigdc/components/DownloadVisualizationButton';
+import wrapSvg from '@ncigdc/utils/wrapSvg';
+import {
+  MAXIMUM_CURVES,
+  MINIMUM_CASES,
+} from '@ncigdc/utils/survivalplot';
+import { SpinnerIcon, CloseIcon,
+  SurvivalIcon,
+  BarChartIcon,
+  BoxPlot, } from '@ncigdc/theme/icons';
+import termCapitaliser from '@ncigdc/utils/customisation';
+import {
+  humanify,
+} from '@ncigdc/utils/string';
+import {
+  removeClinicalAnalysisVariable,
+  updateClinicalAnalysisVariable,
+} from '@ncigdc/dux/analysis';
+
+import ActionsDropdown from './components/ActionsDropdown';
+import ClinicalBoxPlot from './components/ClinicalBoxPlot';
+import ClinicalHistogram from './components/ClinicalHistogram';
+import ClinicalSurvivalPlot from './components/ClinicalSurvivalPlot';
+
+import {
+  boxTableAllowedStats,
+  boxTableRenamedStats,
+  colors,
+  getCardFilters,
+  getHeadings,
+  parseContinuousValue,
+  styles,
+  // vizButtons,
+} from './helpers';
+
+const vizButtons = {
+  box: {
+    action: updateClinicalAnalysisVariable,
+    icon: <BoxPlot style={styles.chartIcon} />,
+    title: 'Box/QQ Plot',
+  },
+  delete: {
+    action: removeClinicalAnalysisVariable,
+    icon: <CloseIcon style={styles.chartIcon} />,
+    title: 'Remove Card',
+  },
+  histogram: {
+    action: updateClinicalAnalysisVariable,
+    icon: <BarChartIcon style={styles.chartIcon} />,
+    title: 'Histogram',
+  },
+  survival: {
+    action: updateClinicalAnalysisVariable,
+    icon: <SurvivalIcon style={styles.chartIcon} />,
+    title: 'Survival Plot',
+  },
+};
+
+const getTableData = (
+  displayData,
+  fieldName,
+  getContinuousBins,
+  selectedBins,
+  selectedSurvivalBins,
+  selectedSurvivalLoadingIds,
+  setId,
+  setSelectedBins,
+  totalDocs,
+  updateSelectedSurvivalBins,
+  variable,
+) => (displayData.length === 0
```

Comment:
```suggestion
) => (Array.isArray(displayData)
```
or use typescript to declare its type.

Reply:
I'm checking for an empty array here, so `length === 0` is the most accurate check

We should have some type checking going on though

Reply:
Not sure why you need to check an empty array here, since you return displayData.map(...). Even if displayData is an empty array, .map won't change anything and return the empty array. Beside, the displayData can be undefined or null, which will give you  `undefined` error.

Reply:
talked to Justin and came up with

```js
const getTableData = (
displayData = [], // set default value
....
) => (displayData.length > 0 ? displayData.map(...) : displayData);
```

Reply:
Sure. But sorry I still don't know why need to check if it's an empty array. [].map(...) will return []. No need to have a condition here.

Reply:
This is true :+1:

Reply:
Well, that's to avoid calling the method unnecessarily...

Reply:
> Well, that avoids calling the method unnecessarily... saving that bit of processing time.

That's so tiny. Beside, use `if` everytime also cost processing time, although it's  also tiny.